### PR TITLE
feat: color updates to the design tokens package

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This PR introduces updates to the Figma, CSS, and JS design tokens for improved accessibility, consistency, and usability across both Light and Dark themes. Key changes include updates to border and icon colors, the addition of new background color tokens, and alignment of descriptions with the latest in Figma. Migrated from https://github.com/MetaMask/design-tokens/pull/741

### **Light Theme Updates**

- **Border:**
  - Updated `border.default` from `{grey.200}` to `{grey.400}` for improved accessibility.

- **Icons:**
  - Linked icons to text color tokens:
    - `icon.default`: `{grey.900}` → `{text.default}`
    - `icon.alternative`: `{grey.500}` → `{text.alternative}`
    - `icon.muted`: `{grey.300}` → `{text.muted}`

- **New Colors Added:**
  - `background-muted`: `{grey.050}`
  - `background-muted-hover`: `#E7EBEE`
  - `background-muted-pressed`: `#DBE0E6`

- **Descriptions:**
  - Updated most descriptions to match the latest designs in Figma.

---

### **Dark Theme Updates**

- **Background:**
  - Darkened defaults for better contrast:
    - `background-default`: `{grey.800}` → `{grey.900}`
    - `background-alternative`: `{grey.900}` → `{grey.1000}`
  - Updated hover and pressed states:
    - `background-default-hover`: `#313235` → `#1E2124`
    - `background-default-pressed`: `#3F4145` → `#272B2F`
    - `background-alternative-hover`: `#1F2123` → `#0A0A0A`
    - `background-alternative-pressed`: `#2E3033` → `#141414`

- **Border:**
  - Adjusted `border.default` from `{grey.400}` to `{grey.500}` to align with Light theme contrast levels.

- **Text:**
  - Improved contrast for better readability:
    - `text.alternative`: `{grey.200}` → `{grey.300}`
    - `text.muted`: `{grey.400}` → `{grey.500}`

- **Icons:**
  - Linked icons to text color tokens:
    - `icon.default`: `{grey.000}` → `{text.default}`
    - `icon.alternative`: `{grey.300}` → `{text.alternative}`
    - `icon.muted`: `{grey.500}` → `{text.muted}`

- **New Colors Added:**
  - `background-muted`: `{grey.800}`
  - `background-muted-hover`: `#2D3034`
  - `background-muted-pressed`: `#363B3F`

- **Descriptions:**
  - Updated most descriptions to match the latest designs in Figma.

## **Related Issues**

Fixes: https://github.com/MetaMask/metamask-design-system/issues/181

## **Manual Testing Steps**

1. **Run Storybook:**
   - Execute `yarn storybook` in your terminal.
   - Navigate to the **Design Tokens > Theme Colors** section.
   - Verify updates for both **Light** and **Dark** themes:
     - Updated color values match the PR specifications.
     - Descriptions align with the latest Figma updates.
2. Test new color tokens updates in mobile, extension and portfolio
3. Validate **VSCode Intellisense** for JS tokens:
  
### **After**
Updated colors are reflected in storybook for Figma, JSinCSS and CSS theme colors

https://github.com/user-attachments/assets/8cfead77-4fe7-4691-aee3-88c5dffc7b48

## Mobile
Mobile: Design token updates working as expected for light and dark mode

### Before

https://github.com/user-attachments/assets/e94144ba-777d-4c1e-8b1c-ef5735a0bc07

### After

https://github.com/user-attachments/assets/5d989a94-5379-470d-926f-3aef1a92c9b1

## Extension
Extension: Design token updates working as expected for light and dark mode

### Before
https://github.com/user-attachments/assets/8599f456-32ca-44b2-bafb-cc0e037e4ad3

### After
https://github.com/user-attachments/assets/25121ae5-de06-4f99-80dd-187c0ea67744

## Portfolio
Portfolio: Design token updates working as expected for light and dark mode

### Before

https://github.com/user-attachments/assets/4f1ae0a0-fcdb-4e82-8925-fbc768c2ba78

### After

https://github.com/user-attachments/assets/7385ce6f-625b-415f-b6cb-29d962d689bf

### **After**
Intellisense working for theme types

https://github.com/user-attachments/assets/6a8a0270-ba76-4b44-bb5f-ebb208316590

## **Pre-merge Checklist**

- [x] Followed [[MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md)](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] Updated documentation or references to token usage if needed.
- [x] Included appropriate tests for changes.
- [x] Applied relevant labels.

### CODE REVIEW

File: packages/design-tokens/stories/Shadows.stories.tsx [hunk 1]
```diff
@@ -3,10 +3,10 @@ import type { Meta, StoryObj } from '@storybook/react';
 import { Text, TextColor } from '@metamask/design-system-react';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Shadows.stories.tsx:3]: Move to new PR

File: packages/design-tokens/stories/components/ColorSwatch/ColorSwatch.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, { FunctionComponent } from 'react';
 import { lightTheme } from '../../../src';
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/ColorSwatch/ColorSwatch.tsx:2]: Move to new PR and update tsconfig to catch these in stories

File: packages/design-tokens/stories/components/ColorSwatchGroup/ColorSwatchGroup.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { getContrastYIQ } from '../../test-utils/getContrastYIQ';
 import { ColorSwatch } from '..';
 import { Text, TextVariant } from '@metamask/design-system-react';
 
-interface ColorSwatchGroupProps {
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/components/ColorSwatchGroup/ColorSwatchGroup.tsx:7]: Move to separate PR

File: apps/storybook-react-native/.storybook/storybook.requires.js [hunk 4]
```diff
@@ -6,30 +6,30 @@ import {
   addParameters,
```
Comment (georgewrmarshall) [inline apps/storybook-react-native/.storybook/storybook.requires.js:6]: Move to another PR

File: packages/design-tokens/stories/Typography.stories.tsx [hunk 5]
```diff
@@ -531,9 +531,7 @@ export const SmallScreenCSS: StoryFn<typeof Text> = (...args) => {
               value.lineHeight,
```
Comment (georgewrmarshall) [inline packages/design-tokens/stories/Typography.stories.tsx:531]: Move to separate PR

File: packages/design-system-tailwind-preset/src/colors.ts [hunk 6]
```diff
@@ -6,6 +6,9 @@ export const colors = {
     alternative: 'var(--color-background-alternative)',
     'alternative-hover': 'var(--color-background-alternative-hover)',
     'alternative-pressed': 'var(--color-background-alternative-pressed)',
+    muted: 'var(--color-background-muted)',
+    'muted-hover': 'var(--color-background-muted-hover)',
+    'muted-pressed': 'var(--color-background-muted-pressed)',
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/colors.ts:11]: Adding the new `background-muted` to the Tailwind preset. A test initially failed when the new `background-muted` CSS variables were added to the design tokens, which is great!  

Screenshot 2024-11-29 at 9 03 29 AM

**Comment by georgewrmarshall** [comment]
@metamaskbot publish-preview

File: packages/design-tokens/src/js/themes/types.ts [hunk 7]
```diff
@@ -23,225 +23,133 @@ export type ThemeShadows = {
...
     inverse: string;
   };
+  shadow: {
+    /** For neutral drop shadow color */
+    default: string;
+    /** For primary drop shadow color */
+    primary: string;
+    /** For critical/danger drop shadow color */
+    error: string;
+  };
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/types.ts:152]: Adding missing shadow type

File: packages/design-tokens/src/js/themes/types.ts [hunk 8]
```diff
@@ -23,225 +23,133 @@ export type ThemeShadows = {
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/types.ts:23]: Updating descriptions to match Figma

File: packages/design-tokens/src/js/themes/lightTheme/shadows.test.ts [hunk 9]
```diff
@@ -7,27 +7,27 @@ const designTokens = require('../../../figma/tokens.json');
 describe('Shadows', () => {
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/lightTheme/shadows.test.ts:7]: Fixing tests

File: packages/design-tokens/src/js/themes/lightTheme/lightTheme.test.ts [hunk 10]
```diff
@@ -8,7 +8,9 @@ describe('Light Theme', () => {
   it('color tokens are exported from lightTheme by checking a random color token', () => {
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/lightTheme/lightTheme.test.ts:8]: Fixing tests

File: packages/design-tokens/src/js/themes/lightTheme/colors.ts [hunk 11]
```diff
@@ -3,76 +3,131 @@ import type { ThemeColors } from '../types';
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/lightTheme/colors.ts:3]: Improving comments and aligning to Figma

File: packages/design-tokens/src/js/themes/lightTheme/colors.test.ts [hunk 12]
```diff
@@ -30,11 +30,13 @@ function resolveColorReferences(
       if (match) {
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/lightTheme/colors.test.ts:30]: Improving tests

File: packages/design-tokens/src/js/themes/darkTheme/darkTheme.test.ts [hunk 13]
```diff
@@ -8,7 +8,9 @@ describe('dark Theme', () => {
   it('color tokens are exported from darkTheme by checking a random color token', () => {
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/darkTheme/darkTheme.test.ts:8]: Improving tests

File: packages/design-tokens/src/js/themes/darkTheme/colors.ts [hunk 14]
```diff
@@ -3,76 +3,131 @@ import type { ThemeColors } from '../types';
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/darkTheme/colors.ts:3]: Improving comments and aligning with updated Figma comments

File: packages/design-tokens/src/js/themes/darkTheme/colors.test.ts [hunk 15]
```diff
@@ -30,11 +30,13 @@ function resolveColorReferences(
       if (match) {
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/js/themes/darkTheme/colors.test.ts:30]: Improving tests

File: packages/design-tokens/src/figma/lightTheme.json [hunk 16]
```diff
@@ -4,283 +4,301 @@
       "value": "{grey.000}",
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/figma/lightTheme.json:4]: Updating comments and adding background muted colors

File: packages/design-tokens/src/figma/darkTheme.json [hunk 17]
```diff
@@ -1,286 +1,304 @@
 {
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/figma/darkTheme.json:1]: Updating comments and adding background muted

File: packages/design-tokens/src/css/lightTheme.test.ts [hunk 18]
```diff
@@ -26,11 +26,13 @@ describe('Light Theme Colors CSS', () => {
               throw new Error(`Invalid color or shade: ${value as string}`);
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/css/lightTheme.test.ts:26]: Improving tests

File: packages/design-tokens/src/css/light-theme-colors.css [hunk 19]
```diff
@@ -4,102 +4,108 @@
 :root,
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/css/light-theme-colors.css:4]: Updating comments to align with figma and adding background muted

File: packages/design-tokens/src/css/darkTheme.test.ts [hunk 20]
```diff
@@ -26,11 +26,13 @@ describe('Dark Theme Colors CSS', () => {
               throw new Error(`Invalid color or shade: ${value as string}`);
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/css/darkTheme.test.ts:26]: Improving tests

File: packages/design-tokens/src/css/dark-theme-colors.css [hunk 21]
```diff
@@ -3,102 +3,108 @@
 */
```
Comment (georgewrmarshall) [inline packages/design-tokens/src/css/dark-theme-colors.css:3]: Updating comments and colors as well as adding background muted

File: packages/design-tokens/src/js/themes/darkTheme/colors.ts [hunk 22]
```diff
@@ -3,76 +3,131 @@ import type { ThemeColors } from '../types';
...
-    default: brandColor.grey800, // For default neutral backgrounds
-    alternative: brandColor.grey900, // For secondary neutral backgrounds.
-    defaultHover: '#313235', // For "hover" states that use background-default.
-    defaultPressed: '#3f4145', // For "pressed" states that use background-alternative.
-    alternativeHover: '#1f2123', // For "hover" states that use background-alternative.
-    alternativePressed: '#2e3033', // For "pressed" states that use background-alternative.
-    hover: '#ffffff0a', // For "hover" state that use no background fill.
-    pressed: '#ffffff14', // For "pressed" state that use no background fill.
+    /** For default neutral surface (#24272A) */
+    default: brandColor.grey900,
```
Comment (brianacnguyen) [inline packages/design-tokens/src/js/themes/darkTheme/colors.ts:7]: yaaas

**Comment by Akatori-Design** [comment]
Approved!
